docker/dockerfile

```diff
@@ -2,15 +2,41 @@ ARG BASE_IMAGE
 
 FROM ${BASE_IMAGE} AS requirements
 
+ARG PYTHON_VERSION=3.12
+ENV VIRTUAL_ENV=/opt/venv-${PYTHON_VERSION}
+ENV PATH=${VIRTUAL_ENV}/bin:${PATH}
+
+RUN <<EOF
+    set -ex
+    apt update
+    apt --yes --no-install-recommends install python3 python3-pip python3-venv git
+
+    printf "import re, sys\nif re.match(r'Python ${PYTHON_VERSION}.[0-9]+', '$(python3 --version)') is None: raise RuntimeError(f'Your Python version {sys.version} does not match ${PYTHON_VERSION}.')" | python3
+
+    python3 -m venv $VIRTUAL_ENV
+    apt clean && rm -rf /var/lib/apt/lists/*
+EOF
+
+ARG COMPILER_FAMILY
+
 # Install APT requirements.
 RUN --mount=target=/requirements,type=bind,source=requirements <<EOF
     set -ex
 
-    apt update
+    python3 -m pip install git+https://github.com/uliegecsm/system-helpers
 
-    apt --yes --no-install-recommends install $(grep -vE "^\s*#" /requirements/requirements.system.txt  | tr "\n" " ")
+    if [ "${COMPILER_FAMILY}" = "gcc" ];then
+        COMPILER_FAMILY_REQUIREMENTS=g++
+    elif [ "${COMPILER_FAMILY}" = "clang" ];then
+        COMPILER_FAMILY_REQUIREMENTS="clang clang-tidy libomp-dev"
```

Comment:
Just wondering if we need `libstdc++-dev` for the `gcc` standard library in this case?

Reply:
Not needed in this case, we use defaulted versions of everything.